### PR TITLE
Allow urls to new frontend views to link to the public facing site on internal instances

### LINKED ISSUE #10862: Trying to access a public profile from django admin Users will result in a 500 error
This looks like a regression from https://github.com/mozilla/addons-server/pull/10833

STR:
1. Log in to AMO -dev with an admin account
2. Access the User profiles management tool
3. Open the details of a user from the list - i.e. https://addons-internal.dev.mozaws.net/en-US/admin/models/users/userprofile/10641829/change/
4. Click on `View on site` button on the top right corner

Actual result:
A server error (500) is received

Expected result:
The public profile of the user is opened

Notes:
- the link should be pointing to the new frontend profile page
- reproduced on AMO -dev with FF65, Win10x64

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #10862 and #11014 and any other place /admin links to a frontend view.

### CODE REVIEW

**Comment by eviljeff** [comment]
Actually, still not sure about my approach here.   Maybe I should be just adding handling for an external url directly in each model's `get_url_path`

**Comment by EnTeQuAk** [comment]
Yes, I personally think that a models `get_url_path` is the better approach.

This gives us much more freedom I think.

**Comment by eviljeff** [comment]
Although, the negative of that approach is duplicating the same logic in ~8 different model's `get_url_path`

**Comment by diox** [comment]
Maybe we could have a generic implementation in `ModelBase` ? I really don't like touching `frontend_view` either.

**Comment by diox** [comment]
Note: the API already has to deal with something similar. Serializers all wrap `get_url_path()` in a `absolutify()` call. Maybe there is something to refactor there so that the admin etc also do this.

**Comment by eviljeff** [comment]
Re-implemented how this works to replace `get_absolute_url` instead.  A follow-up would be to replace the `absolutify(instance.get_url_path()` in the API serailizers.

File: src/olympia/reviewers/views.py [hunk 1]
```diff
@@ -1201,8 +1202,9 @@ def download_git_stored_file(request, version_id, filename):
         raise http.Http404()
 
     actual_blob = serializer.git_repo[blob_or_tree.oid]
+
     response = http.FileResponse(
-        memoryview(actual_blob),
+        io.BytesIO(memoryview(actual_blob)),
```
Comment (EnTeQuAk) [inline src/olympia/reviewers/views.py]: That shouldn't be in here.

File: src/olympia/reviewers/tests/test_views.py [hunk 2]
```diff
@@ -5682,6 +5686,10 @@ def _test_url_success(self):
         response = self.client.get(url)
         assert response.status_code == 200
 
+        content = b''.join(response.streaming_content).decode('utf-8')
+        assert content.startswith('{')
+        assert '"manifest_version": 2' in content
```
Comment (EnTeQuAk) [inline src/olympia/reviewers/tests/test_views.py]: That shouldn't be in here.

File: src/olympia/reviewers/tests/test_views.py [hunk 3]
```diff
@@ -5637,6 +5637,10 @@ def test_download_basic(self):
             response['Content-Disposition'] ==
             'attachment; filename="manifest.json"')
 
+        content = b''.join(response.streaming_content).decode('utf-8')
+        assert content.startswith('{')
+        assert '"manifest_version": 2' in content
```
Comment (EnTeQuAk) [inline src/olympia/reviewers/tests/test_views.py]: That shouldn't be in here.

File: src/olympia/reviewers/views.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/reviewers/views.py]: rebased and force pushed to get rid of the extra commit

**Pull request merged**